quickjs-libc.c

```diff
@@ -2171,9 +2180,9 @@ static int js_os_run_timers(JSRuntime *rt, JSContext *ctx, JSThreadState *ts)
                 th->timeout = cur_time + th->delay;
             else
                 free_timer(rt, th);
-            call_handler(ctx, func);
+            r = call_handler(ctx, func);
             JS_FreeValueRT(rt, func);
-            return 0;
+            return r;
```

Comment:
Rather than making callers check for an out-of-band `JS_IsException(ts->exc)` (easy to forget), it's more robust and (arguably) idiomatic to turn min_delay into an out param and make js_os_run_timers return 0 or -1.

Reply:
Cna do!